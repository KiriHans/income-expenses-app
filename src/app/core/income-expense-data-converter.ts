import {
  FirestoreDataConverter,
  QueryDocumentSnapshot,
  SnapshotOptions,
} from '@angular/fire/firestore';
import { IncomeExpense } from './models/income-expenses.model';

export const incomeExpenseConverter: FirestoreDataConverter<IncomeExpense> = {
  toFirestore: (incomeExpense: IncomeExpense) => {
    const { description, amount, type, uid } = incomeExpense;
    return { description, amount, type, uid };
  },
  fromFirestore: (
    snapshot: QueryDocumentSnapshot<IncomeExpense>,
    options: SnapshotOptions | undefined
  ) => {
    const data = snapshot.data(options);
    return new IncomeExpense(data.description, data.amount, data.type, data.uid);
  },
};
