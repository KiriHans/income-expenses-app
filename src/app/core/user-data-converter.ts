import {
  FirestoreDataConverter,
  QueryDocumentSnapshot,
  SnapshotOptions,
} from '@angular/fire/firestore';
import { UserApp } from './models/users.model';

export const userConverter: FirestoreDataConverter<UserApp> = {
  toFirestore: (user: UserApp) => {
    const { displayName, email } = user;
    return { displayName, email };
  },
  fromFirestore: (
    snapshot: QueryDocumentSnapshot<UserApp>,
    options: SnapshotOptions | undefined
  ) => {
    const data = snapshot.data(options);
    return new UserApp(data.id, data.displayName, data.email);
  },
};
