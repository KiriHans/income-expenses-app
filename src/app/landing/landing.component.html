<main class="page-body-wrapper flex-column  full-page-wrapper h-100  p-0 ">

    <app-section>
        <div class="d-flex row gap-md-5 gap-4 align-items-center justify-content-center vh-100">
            <div class="col-md-6 col-12">
                <h1 class="fs-md-6 display-2 lh-1 pb-2">
                    Keep an eye of your finances without effort!
                </h1>
                <h2 class="lead">
                    <small class="text-body-secondary">
                        With our Income Expense App, managing your money has never been easier. Keep track of your
                        financial transactions, monitor your income, and control your expenses effortlessly.
                    </small>
                </h2>
                <div class="d-grid gap-2 mt-4 w-25 ">
                    <button routerLink="/auth/login" class="btn btn-primary btn-lg fs-5">Get started</button>
                </div>


            </div>
            <div class="col d-flex align-items-center justify-content-center">
                <figure class="figure">
                    <img class="figure-img img-fluid rounded" alt="Brainstorming over paper" loading="eager"
                        sizes="(min-width: 1000px) 710px, 360px"
                        srcset="/assets/images/landing/website-360.png 360w, /assets/images/landing/website-710.png 710w, /assets/images/landing/website.png 1092w">

                </figure>
            </div>
        </div>
    </app-section>






</main>