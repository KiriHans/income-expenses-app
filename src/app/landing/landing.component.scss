.decoration-1{
    position: absolute;
    right: 10%;
    bottom: -20%;
    transform: rotate(45deg);
}

