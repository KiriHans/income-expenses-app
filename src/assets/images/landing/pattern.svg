<svg xmlns="http://www.w3.org/2000/svg" version="1.1" xmlns:xlink="http://www.w3.org/1999/xlink"
    xmlns:svgjs="http://svgjs.dev/svgjs" class="pattern" viewBox="0 0 400 400">
    <g class="shape-0">
        <rect width="100" height="100" fill="#939473" x="0" y="0"></rect>
        <path d="M 0 0 L 100 0 L 100 100 Z"
            transform="matrix(-1,-1.2246467991473532e-16,1.2246467991473532e-16,-1,100,100)"
            fill="#13444d"></path>
    </g>
    <g class="shape-1">
        <rect width="100" height="100" fill="#939473" x="0" y="100"></rect>
        <circle r="25" cx="50" cy="150" fill="#4f784e"></circle>
    </g>
    <g class="shape-2">
        <rect width="100" height="100" fill="#939473" x="0" y="200"></rect>
        <circle r="25" cx="50" cy="250" fill="#13444d"></circle>
    </g>
    <g class="shape-3">
        <rect width="100" height="100" fill="#2d5e4c" x="0" y="300"></rect>
        <path d="M 0 300 L 100 300 L 100 400 Z"
            transform="matrix(-1,-1.2246467991473532e-16,1.2246467991473532e-16,-1,99.99999999999996,700)"
            fill="#252326"></path>
    </g>
    <g class="shape-4">
        <rect width="100" height="100" fill="#2d5e4c" x="100" y="0"></rect>
        <circle r="25" cx="150" cy="50" fill="#4f784e"></circle>
    </g>
    <g class="shape-5">
        <rect width="100" height="100" fill="#13444d" x="100" y="100"></rect>
        <path d="M100 100L150 100A50 50 0 0 1 100 150Z " fill="#939473"></path>
        <path d="M50 50L100 50A50 50 0 0 1 50 100Z "
            transform="matrix(-1,1.2246467991473532e-16,-1.2246467991473532e-16,-1,250,250)"
            fill="#939473"></path>
    </g>
    <g class="shape-6">
        <rect width="100" height="100" fill="#939473" x="100" y="200"></rect>
        <path d="M 100 200 L 200 200 L 200 300 Z"
            transform="matrix(-1,-1.2246467991473532e-16,1.2246467991473532e-16,-1,300,500)"
            fill="#13444d"></path>
    </g>
    <g class="shape-7">
        <rect width="100" height="100" fill="#2d5e4c" x="100" y="300"></rect>
        <path d="M 100 300 L 200 300 L 200 400 Z"
            transform="matrix(-1,-1.2246467991473532e-16,1.2246467991473532e-16,-1,299.99999999999994,700)"
            fill="#939473"></path>
    </g>
    <g class="shape-8">
        <rect width="100" height="100" fill="#4f784e" x="200" y="0"></rect>
        <circle r="25" cx="250" cy="50" fill="#13444d"></circle>
    </g>
    <g class="shape-9">
        <rect width="100" height="100" fill="#4f784e" x="200" y="100"></rect>
        <circle r="25" cx="250" cy="150" fill="#939473"></circle>
    </g>
    <g class="shape-10">
        <rect width="100" height="100" fill="#939473" x="200" y="200"></rect>
        <path d="M200 200L250 200A50 50 0 0 1 200 250Z " fill="#4f784e"></path>
        <path d="M150 150L200 150A50 50 0 0 1 150 200Z "
            transform="matrix(-1,1.2246467991473532e-16,-1.2246467991473532e-16,-1,450,450)"
            fill="#4f784e"></path>
    </g>
    <g class="shape-11">
        <rect width="100" height="100" fill="#4f784e" x="200" y="300"></rect>
        <circle r="25" cx="250" cy="350" fill="#13444d"></circle>
    </g>
    <g class="shape-12">
        <rect width="100" height="100" fill="#2d5e4c" x="300" y="0"></rect>
        <path d="M 300 0 L 400 0 L 400 100 Z"
            transform="matrix(-1,-1.2246467991473532e-16,1.2246467991473532e-16,-1,700,100.00000000000004)"
            fill="#939473"></path>
    </g>
    <g class="shape-13">
        <rect width="100" height="100" fill="#4f784e" x="300" y="100"></rect>
        <circle r="25" cx="350" cy="150" fill="#939473"></circle>
    </g>
    <g class="shape-14">
        <rect width="100" height="100" fill="#4f784e" x="300" y="200"></rect>
        <path d="M 300 200 L 400 200 L 400 300 Z"
            transform="matrix(-1,-1.2246467991473532e-16,1.2246467991473532e-16,-1,700,500.00000000000006)"
            fill="#2d5e4c"></path>
    </g>
    <g class="shape-15">
        <rect width="100" height="100" fill="#4f784e" x="300" y="300"></rect>
        <path d="M 300 300 L 400 300 L 400 400 Z"
            transform="matrix(-1,-1.2246467991473532e-16,1.2246467991473532e-16,-1,700,700)"
            fill="#252326"></path>
    </g>
</svg>