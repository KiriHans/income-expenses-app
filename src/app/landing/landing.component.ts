import { Component } from '@angular/core';
import { RouterLink } from '@angular/router';
import { NavbarLandingComponent } from '../shared/components/navbar-landing/navbar-landing.component';

@Component({
  selector: 'app-landing',
  standalone: true,
  imports: [RouterLink, NavbarLandingComponent],
  templateUrl: './landing.component.html',
})
export class LandingComponent {}
