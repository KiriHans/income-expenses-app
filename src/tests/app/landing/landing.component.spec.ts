import { DebugElement } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { By } from '@angular/platform-browser';

import { LandingComponent } from 'src/app/landing/landing.component';

describe('LandingComponent', () => {
  let component: LandingComponent;
  let fixture: ComponentFixture<LandingComponent>;

  let el: DebugElement;

  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [LandingComponent],
    }).compileComponents();

    fixture = TestBed.createComponent(LandingComponent);
    component = fixture.componentInstance;
    el = fixture.debugElement;

    compiled = fixture.nativeElement;

    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    expect(compiled).toMatchSnapshot();
  });

  it('should countain a title', () => {
    const title = el.query(By.css('h1'));
    expect(title.nativeElement.textContent).toBe('Keep an eye of your finances without effort!');
  });
});
