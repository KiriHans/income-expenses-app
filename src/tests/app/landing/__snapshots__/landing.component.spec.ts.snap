// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingComponent should match snapshot 1`] = `
<div
  id="root1"
>
  <main
    class="page-body-wrapper flex-column full-page-wrapper h-100 p-0"
  >
    <app-section>
      <div
        class="px-4 py-3 flex-grow-1"
      >
        <div
          class="container d-flex align-items-center"
        >
          <div
            class="d-flex row gap-md-5 gap-4 align-items-center justify-content-center vh-100"
          >
            <div
              class="col-md-6 col-12"
            >
              <h1
                class="fs-md-6 display-2 lh-1 pb-2"
              >
                 Keep an eye of your finances without effort! 
              </h1>
              <h2
                class="lead"
              >
                <small
                  class="text-body-secondary"
                >
                   With our Income Expense App, managing your money has never been easier. Keep track of your financial transactions, monitor your income, and control your expenses effortlessly. 
                </small>
              </h2>
              <div
                class="d-grid gap-2 mt-4 w-25"
              >
                <button
                  class="btn btn-primary btn-lg fs-5"
                  routerlink="/auth/login"
                  tabindex="0"
                >
                  Get started
                </button>
              </div>
            </div>
            <div
              class="col d-flex align-items-center justify-content-center"
            >
              <figure
                class="figure"
              >
                <img
                  alt="Brainstorming over paper"
                  class="figure-img img-fluid rounded"
                  loading="eager"
                  sizes="(min-width: 1000px) 710px, 360px"
                  srcset="/assets/images/landing/website-360.png 360w, /assets/images/landing/website-710.png 710w, /assets/images/landing/website.png 1092w"
                />
              </figure>
            </div>
          </div>
        </div>
      </div>
    </app-section>
  </main>
</div>
`;
